compiler/rustc_middle/src/query/mod.rs

```diff
@@ -121,6 +122,21 @@ rustc_queries! {
         desc { "perform lints prior to macro expansion" }
     }
 
+    /// Tracked access to environment variables.
+    ///
+    /// Useful for the implementation of `std::env!`, `proc-macro`s change
+    /// detection and other changes in the compiler's behaviour that is easier
+    /// to control with an environment variable than a flag.
+    ///
+    /// NOTE: This currently does not work with dependency info in the
+    /// analysis, codegen and linking passes, place extra code at the top of
+    /// `rustc_interface::passes::write_dep_info` to make that work.
+    query env_var(key: &'tcx OsStr) -> Option<&'tcx OsStr> {
```

Comment:
```suggestion
    query env_var(key: &OsStr) -> Option<&'tcx OsStr> {
```
Does the key have to be `'tcx`?
The implementation logic doesn't require it.

If the query infra requires it, then it's better to make a wrapper that allocates it on `tcx` arena, otherwise it will be inconvenient to use it.

Reply:
> Does the key have to be `'tcx`?

Yup.

> it's better to make a wrapper that allocates it on `tcx` arena, otherwise it will be inconvenient to use it.

In most cases, the key will be `'static` though.